react-exercise/src/App.tsx

```diff
@@ -1,48 +1,32 @@
 import { useEffect, useState } from "react";
 import { BrowserRouter as Router, Route, Routes } from "react-router-dom";
-import { setUsers } from "./stores/usersStore.ts";
 import UsersDetails from "./pages/usersDetails/UsersDetails.tsx";
 import UserProfile from "./pages/userProfile/UserProfile.tsx";
+import { fetchUsersSync } from "./usersClientService.ts";
 import { FetchState } from "./types/fetchUsersState";
 import "./App.css";
 import { enableMapSet } from "immer";
 
 // Enables the use of Map and Set in Immer
 enableMapSet();
-
 function App() {
   const [fetchState, setFetchState] = useState<FetchState>({
     type: "progress",
   });
 
   useEffect(() => {
-    setFetchState({ type: "progress" });
-    fetch("http://localhost:4000/users")
-      .then((res) => {
-        if (!res.ok) {
-          throw new Error("Failed to fetch users");
-        }
-        return res.json();
-      })
-      .then((data) => {
-        setUsers(data);
-        setFetchState({ type: "success" });
-      })
-      .catch((error) => {
-        console.error("Error fetching users:", error);
-        setFetchState({
-          type: "failure",
-          errorMessage: "Failed to fetch users. Please try again later.",
-        });
-      });
+    fetchUsersSync(
```

Comment:
It's a good idea to set the function params as an object, it make it clearer when using the function, which parameter is which. It's not a must, just makes it easier not to get confused - for example here, it would be easy to pass the progress  and the success blocks.